## Exercises Solutions

-----

**1).** Galvanize decides to do random drug tests for marijuana on their employees. They know that 3% of their population uses marijuana. The drug test has the following accuracy: The test correctly identifies 95% of the marijuana users and 90% of the non-users.

|                | Uses marijuana | Doesn't use marijuana |
|----------------|------------:|-------------------:|
| Tests positive |        0.95 |               0.10 |
| Tests negative |        0.05 |               0.90 |

Jared gets tested and the test comes back positive. What is the probability that he uses marijuana?
   
\begin{align*}
P(\text{uses marijuana } | \text{ tests positive}) 
&= \frac{P(\text{tests positive } | \text{ uses marijuana})P(\text{uses marijuana})}{P(\text{tests positive})}
\\
&= \frac{0.95*0.03}{0.03*.95 + ((1-.03)*0.10)}
\\
&\approx 0.227
\end{align*}



__2__)

<center><img src="../images/example.png" height="500"/></center>

__3__)


$M_{1}$: The coin is absolutely fair, and we are certain of it. Thus we model this by $P(H) = 0.5$, so that the Bernoulli parameter $\theta = \frac{1}{2}$:

$$P(M_{1}) = {{n}\choose{k}}\theta^{n} $$

$M_{2}$: The coin may not be fair. We need to determine the parameters based on a uniform prior parameter $\theta \in \left[0,1\right]$

$$P(M_{2}) = \int_{0}^{1}{{n}\choose{k}}\theta^{k}(1-\theta)^{n-k}d\theta $$

Suppose we flip the coin 100 times and get 61 heads. Calculating the posteriors for each:

$$M_{1} = {{100}\choose{61}}{\frac{1}{2}}^{100} = 0.0071$$

$$M_{2} = {{100}\choose{61}}\int_{0}^{1}{q}^{61}(1-q)^{39}dq$$

$$= \dfrac{\Gamma(m+1)\Gamma(n-m+1)}{\Gamma(n+2)} = \dfrac{n!}{m!(n-m)!}\cdot\dfrac{m!(n-m)!}{(n+1)!} $$

$$= \dfrac{n!}{m!(n-m)!}\cdot\dfrac{m!(n-m)!}{(n+1)!} = \dfrac{n!}{(n+1)!} = \dfrac{1}{n+1}$$

$$M_{2} = 0.01$$

The Bayes factor is slightly in favor of the second model.

In this case it looks like $M_{2}$ is weakly supported here. 


In [3]:
0.0071 / 0.01 # M2 is better than M1

0.7100000000000001

In [4]:
0.01 / 0.0071 # Flip to see how much better

# Barely worth mentioning

1.408450704225352

1.4 is "Barely worth mentioning"

----
Challenge Exercises
----

----
A GalvanizeU grad is looking for her first job! Given that she is freaked out, her chances of not getting an offer are 70%. Given that she isn't freaked out, her chances of not getting an offer are 30%. Suppose that the probability that she's freaked out is 80%. What's the probability that she gets an offer?


\begin{align*}
P(\text{she gets an offer})
&= 1 - P(\text{no offer } | \text{ freaked out})P(\text{freaked out}) - P(\text{no offer } | \text{ not freaked out})P(\text{not freaked out})
\\
&= 1 - 0.8*0.7 - 0.2*0.3
\\
&= 0.38
\end{align*}